js/io.ts

```diff
@@ -159,3 +161,31 @@ export function toAsyncIterator(r: Reader): AsyncIterableIterator<Uint8Array> {
     }
   };
 }
+
+/** Read lines asynchronously from `r`.
+ *
+ *      for await (const line of Deno.lines(reader)) {
+ *          console.log(line);
+ *      }
+ */
+export async function* lines(r: Reader): AsyncIterableIterator<string> {
+  const decoder = new TextDecoder();
+  let scratch = "";
+
+  for await (const chunk of toAsyncIterator(r)) {
+    scratch += decoder.decode(chunk);
+
+    let newLineIndex = scratch.indexOf("\n");
```

Comment:
Don't forget about windows carriage return line feeds `\r\n`. Maybe you could keep the code for finding the index of `\n`, but also check the index in scratch before the '\n' and check if it's the `\r` character then skip both the characters if that is the case.

Reply:
Using something like this for the remaining lines too.
```ts
let crLf = false;
let idx = newLineIndex;
if(scratch[idx-1]=="\r") {
  idx--;
  crlf = true;
}
const line = scratch.slice(0, idx);

...

const remainingLines;
if(crLF) {
  remainingLines = scratch.split("\n");
} else {
  remainingLines = scratch.split("\r\n");
}
```

Reply:
@johnsonjo4531 you're right, that's where your implementation is better :) let's wait on `BufReader` it should solve this issue.